Runtime/Aplib/Integrations/Unity/Actions/UnityPathfinderAction.cs

```diff
@@ -0,0 +1,65 @@
+using Aplib.Core;
+using Aplib.Core.Belief;
+using System;
+using UnityEngine;
+using UnityEngine.AI;
+
+namespace Aplib.Integrations.Unity.Actions
+{
+    public class UnityPathfinderAction<TBeliefSet> : Core.Intent.Actions.Action<TBeliefSet>
+        where TBeliefSet : IBeliefSet
+    {
+        public UnityPathfinderAction(
+            Func<TBeliefSet, Rigidbody> objectQuery,
+            Action<TBeliefSet, Vector3> effect,
+            Vector3 location,
+            float heightOffset = 0f,
+            Metadata metadata = null)
+            : this(objectQuery, location: ConstantLocation(location), effect, heightOffset, metadata)
+        {
+        }
+
+        public UnityPathfinderAction(
+            Func<TBeliefSet, Rigidbody> objectQuery,
+            Func<TBeliefSet, Vector3> location,
+            Action<TBeliefSet, Vector3> effect,
+            float heightOffset = 0f,
+            Metadata metadata = null)
+            : base(PathfindingAction(objectQuery, location, effect, heightOffset: heightOffset), metadata)
+        {
+        }
+
+        protected static Func<TBeliefSet, Vector3> ConstantLocation(Vector3 location) => _ => location;
+
+        private static Action<TBeliefSet> PathfindingAction(Func<TBeliefSet, Rigidbody> objectQuery,
+            Func<TBeliefSet, Vector3> locationQuery,
+            Action<TBeliefSet, Vector3> effect,
+            float speed = 7f,
+            float heightOffset = 0f) => beliefSet =>
+        {
+            Rigidbody rigidbody = objectQuery(beliefSet);
+            Vector3 target = locationQuery(beliefSet);
+
+            NavMeshPath path = new();
+            NavMesh.CalculatePath(rigidbody.position,
+                target,
+                NavMesh.AllAreas,
+                path
+            );
+
+            if (path.corners.Length <= 1)
+            {
+                return;
+            }
```

Comment:
Is this old code or introduced here? Because I would not want curly brackets here